core/main/src/firebolt/handlers/provider_registrar.rs

```diff
@@ -296,73 +296,69 @@ impl ProviderRegistrar {
+                        .get_result_properties_schema(event)
+                    {
+                        // Populate the event result, injecting the app ID if the field exists in the event schema
+
+                        let mut result_map = Map::new();
+
+                        for key in event_schema_map.keys() {
+                            if let Some(event_value) = event_data_map.get(key) {
+                                result_map.insert(key.clone(), event_value.clone());
+                            } else if key.eq("appId") {
+                                if let Some(context) = call_context.clone() {
+                                    result_map.insert(key.clone(), Value::String(context.app_id));
+                                } else {
+                                    error!("callback_app_event_emitter: Missing call context, could not determine app ID");
+                                    result_map.insert(key.clone(), Value::Null);
+                                }
                             } else {
-                                error!("callback_app_event_emitter: Missing call context, could not determine app ID");
-                                result_map.insert(key.clone(), Value::Null);
-                            }
-                        } else {
-                            error!(
+                                error!(
                                 "callback_app_event_emitter: Missing field in event data: field={}",
                                 key
                             );
-                            result_map.insert(key.clone(), Value::Null);
+                                result_map.insert(key.clone(), Value::Null);
+                            }
                         }
-                    }
 
-                    if let Some(app_event) = is_app_event {
-                        let app_id = SerdeClearString::as_clear_string(&app_event);
-                        AppEvents::emit_to_app(
-                            &context.platform_state,
-                            app_id,
-                            &FireboltOpenRpcMethod::name_with_lowercase_module(event),
-                            &Value::Object(result_map),
-                        )
-                        .await;
+                        Value::Object(result_map)
                     } else {
-                        AppEvents::emit(
-                            &context.platform_state,
-                            &FireboltOpenRpcMethod::name_with_lowercase_module(event),
-                            &Value::Object(result_map),
-                        )
-                        .await;
+                        event_data.clone()
                     }
-                } else {
-                    error!("callback_app_event_emitter: Result schema not found");
-                    return Err(Error::Custom(String::from("Result schema not found")));
                 }
+                _ => event_data.clone(),
```

Comment:
If the incoming event data is not a Value::Object then just emit whatever it is, no need to check for app ID injection.